lib/scripting/post-process/helpers.ts

```diff
@@ -17,304 +17,103 @@
+	let estree = null;
+	if (node.type === 'Statements' || node.type === 'StatementsInline') {
+		estree = ppStatements(node);
+	} else if (node.type === 'Block') {
+		estree = ppBlock(node);
+	} else if (node.type === 'ArrayTypeModifiers') {
+		estree = ppArrayTypeModifiers(node);
+	} else if (node.type === 'ArraySizeInitializationModifier') {
+		estree = ppArraySizeInitializationModifier(node);
+	} else if (node.type === 'BoundList') {
+		estree = ppBoundList(node);
+	} else if (node.type === 'Identifier' || node.type === 'IdentifierOrKeyword') {
+		estree = ppIdentifier(node);
+	} else if (node.type === 'ArgumentList') {
+		estree = ppArgumentList(node);
 	}
-	return stmt;
-}
-
-export function globalStmtList(result: [Statement, Statement[]]): Statement[] {
-	const stmt = result[0];
-	const otherStmts = result[1] || [];
-	return [stmt, ...otherStmts];
+	return estree;
 }
 
-export function blockStmtList(result: [Statement, BlockStatement]): BlockStatement {
-	const stmt = result[0];
-	const otherBlockStmt = result[1];
-	return estree.blockStatement(otherBlockStmt != null ? [stmt, ...otherBlockStmt.body] : [stmt]);
-}
-
-export function methodStmtList(result: [Statement, Statement[]]): Statement[] {
-	const stmt = result[0];
-	const otherStmts = result[1] || [];
-	return [stmt, ...otherStmts];
-}
-
-export function exitStmt(result: [Token, null, Token]): Statement {
-	const type = result[2].type;
-	return type === 'kw_do' || type === 'kw_for' ? estree.breakStatement() : estree.returnStatement(null);
-}
-
-export function exprList1(result: [Expression, null, Token, null, Expression[]]): Expression[] {
-	const firstExpr = result[0];
-	const otherExprs = result[4] || [];
-	return [firstExpr, ...otherExprs];
-}
-
-export function exprList2(result: [Expression]): Expression[] {
-	const expr = result[0];
-	return [expr];
-}
-
-export function leftExpr1(result: [Identifier, null, Expression[], Token, Expression]) {
-	const identifier = result[0];
-	const indexOrParams = result[2];
-	const expr = result[4];
-	return estree.memberExpression(estree.callExpression(identifier, indexOrParams), expr);
-}
-
-export function leftExpr2(result: [Expression, null, Expression[], Expression]): Expression | null {
-	const identifier = result[0];
-	const indexOrParamsDot = result[2];
-	const leftExprTail = result[3];
-	const callExpr = estree.callExpression(identifier, indexOrParamsDot);
-	if (leftExprTail.type === 'Identifier') {
-		return estree.memberExpression(callExpr, leftExprTail);
-	} else if (leftExprTail.type === 'CallExpression') {
-		return estree.callExpression(
-			estree.memberExpression(callExpr, leftExprTail.callee as Identifier),
-			leftExprTail.arguments as Expression[],
-		);
+function ppStatements(node: ESIToken): any {
+	let stmts: Statement[] = [];
+	for (const child of node.children) {
+		if (!Array.isArray(child.estree)) {
+			stmts.push(child.estree);
+		} else {
+			stmts = stmts.concat(...child.estree);
```

Comment:
Hmm, a simple `stmts.push(...child.estree)` would do, no?

Reply:
Yeh, the first time implemented that I couldn't get spread working. (You could ask my daughter as she was watching me, and seeing my totally baffled) 😃

But fixed!!